home-application/src/main/java/com/softserveinc/ita/homeproject/application/service/QueryApiService.java

```diff
@@ -100,6 +103,27 @@ private static void validateQueryParamValue(String queryParamName, List<String>
         return service.findAll(pageNumber, pageSize, getSpecification(uriInfo));
     }
 
+    /**
+     * @param uriInfo - object that implements UriInfo interface
+     *                  that provides access to application and request URI information
+     * @param service - implementation of QueryableService interface
+     * @return Single DTO
+     * @throws NotFoundHomeException if number of Page elements less than 1
+     * @throws IllegalStateException if number of Page elements more than 1
+     */
+    default D getOne(UriInfo uriInfo, QueryableService<T, D> service) {
+        Page<D> page = getPageFromQuery(uriInfo, service);
+        if (page.getTotalElements() == 1) {
+            return page.getContent().get(0);
+        } else if (page.getTotalElements() == 0) {
+            throw new NotFoundHomeException("Entity with id: " + getParameterValue("id", uriInfo)
```

Comment:
I think that instead of collecting wrong parameters using `getParameterValue("id", uriInfo)` we just can collect it from path parameters. For example:
- query /users/55/contacts/61
- error - "Entity with " + <Use stream or loop to decorate path parameters> + " is not found". ("Entity with 'id = 61 and user_id = 55' is not found")

Reply:
The proposed changes have been made.